### PR TITLE
Implemented Code to take Instance URL from User and use it in place of base URL (#1vrd7g5)

### PR DESCRIPTION
**Pull request by bashu22tiwari**

### CODE REVIEW

File: src/store/modules/user/actions.ts [hunk 1]
```diff
@@ -81,7 +81,15 @@ const actions: ActionTree<UserState, RootState> = {
...
     }
+  },
+
+  /**
+   * Set User Instance Url
+   */
+  setUserInstanceUrl ( { state, commit }, payload){
+    commit(types.USER_INSTANCE_URL,payload)
   }
+
```
Comment (ymaheshwari1) [inline src/store/modules/user/actions.ts:92]: Remove extra lines

File: src/views/Login.vue [hunk 2]
```diff
@@ -48,11 +53,21 @@ export default defineComponent({
...
+      password: "",
+      instanceUrl: ""
     };
   },
+  computed: {
+    ...mapGetters({
+      instanceUrlSaved: 'user/getInstanceUrl'
+    })
+  },
+  mounted() {
```
Comment (ymaheshwari1) [inline src/views/Login.vue:65]: Why we have used mounted hook here? I think we can directly use the getter like `this.instanceUrlSaved` in the template.

File: src/api/index.ts [hunk 3]
```diff
@@ -69,7 +69,9 @@ const api = async (customConfig: any) => {
         data: customConfig.data,
         params: customConfig.params
     }
-    const baseURL = process.env.VUE_APP_BASE_URL;
+
+    let baseURL = store.getters['user/getInstanceUrl'];
+    baseURL = baseURL.match(/^https?:/) ? baseURL : 'https://' + baseURL + '.hotwax.io/api/';
```
Comment (ymaheshwari1) [inline src/api/index.ts:74]: What if I have added an URL like: `https://dev-hc` or `dev-hc.india.co`?

File: src/store/modules/user/actions.ts (hunk 1 above)
Comment (bashu22tiwari) [reply src/store/modules/user/actions.ts]: Removed the unwanted lines.
Thank You for reviewing my code.

File: src/api/index.ts (hunk 3 above)
Comment (bashu22tiwari) [reply src/api/index.ts]: Improved the code to change the base URL according to the output.
Thank You for reviewing my code.

Comment (adityasharma7) [reply src/api/index.ts:74]: Use http

File: src/store/modules/user/mutation-types.ts [hunk 4]
```diff
@@ -2,4 +2,5 @@ export const SN_USER = 'user'
 export const USER_TOKEN_CHANGED = SN_USER + '/TOKEN_CHANGED'
 export const USER_END_SESSION = SN_USER + '/END_SESSION'
 export const USER_INFO_UPDATED = SN_USER + '/INFO_UPDATED'
-export const USER_CURRENT_FACILITY_UPDATED = SN_USER + '/CURRENT_FACILITY_UPDATED'
\ No newline at end of file
+export const USER_CURRENT_FACILITY_UPDATED = SN_USER + '/CURRENT_FACILITY_UPDATED'
+export const USER_INSTANCE_URL = SN_USER + '/INSTANCE_URL'
```
Comment (adityasharma7) [inline src/store/modules/user/mutation-types.ts:6]: Follow the exiiting pattern  
`USER_INSTANCE_URL_UPDATED`

File: src/views/Login.vue [hunk 5]
```diff
@@ -48,11 +53,21 @@ export default defineComponent({
...
+    this.instanceUrl= this.instanceUrlSaved;
+  },
   methods: {
     login: function () {
+      this.store.dispatch("user/setUserInstanceUrl", this.instanceUrl)
```
Comment (adityasharma7) [inline src/views/Login.vue:70]: Use await to avoid any side effects in rare cases

File: src/views/Login.vue [hunk 6]
```diff
@@ -48,11 +53,21 @@ export default defineComponent({
...
     return {
       username: "",
-      password: ""
+      password: "",
+      instanceUrl: ""
     };
   },
+  computed: {
+    ...mapGetters({
+      instanceUrlSaved: 'user/getInstanceUrl'
```
Comment (adityasharma7) [inline src/views/Login.vue:62]: `currentInstanceUrlSaved`

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/api/index.ts (hunk 3 above)
Comment (bashu22tiwari) [reply src/api/index.ts]: Used http instead of https.
Thank You for reviewing my code.

File: src/views/Login.vue (hunk 6 above)
Comment (bashu22tiwari) [reply src/views/Login.vue]: Used the correct name.
Thank You for reviewing my code.

File: src/views/Login.vue (hunk 5 above)
Comment (bashu22tiwari) [reply src/views/Login.vue:70]: This action is working synchronously, so we can avoid using async and await.
Thank You for reviewing my code.

File: src/store/modules/user/mutation-types.ts (hunk 4 above)
Comment (bashu22tiwari) [reply src/store/modules/user/mutation-types.ts]: Updated the name following the existing pattern.
Thank You for reviewing my code.

**Review by adityasharma7 — APPROVED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**